src/js/AppStore.js

```diff
@@ -0,0 +1,196 @@
+        this.setState((state, props) => {
+            return {
+                confirmID: appID,
+                confirmApp: props.apps.find(app => { return app.policyAppID === appID; })
+            }
+        })
+    }
+
+    componentDidMount() {
+        fetch(flags.AppStoreDirectoryUrl).then((res) => {
+            if (!res.ok) {
+                console.error('could not contact app store server');
+                return [];
+            }
+
+            return res.json();
+        }).then((json) => {
+            store.dispatch(updateAvailableAppStoreApps(json.map((app) => {
+                if ('icon_url' in app) {
+                    app.iconUrl = app.icon_url;
+                    delete app.icon_url;
+                }
+                return app;
+            })));
+        });
+    }
+
+    render() {
+        const themeClass = this.props.theme ? 'dark-theme' : 'light-theme';
+        var modalClass = themeClass + " alertOverlay"
+
+        var pendingInstallApp = this.state.confirmApp ? this.state.confirmApp : {
+            name: null,
+            description: null,
+            iconUrl: null
+        };
+
+        return (
+            <div>
+                <Modal
+                isOpen={!!this.state.confirmID}
+                className="alertModal app-body"
+                overlayClassName={modalClass}
+                contentLabel="Example Modal">
+                    <ConfirmAlert appID={this.state.confirmID} name={pendingInstallApp.name}
+                        description={pendingInstallApp.description} iconUrl={pendingInstallApp.iconUrl}
+                        leftText="Cancel" leftCallback={this.cancel} rightText="Download" rightCallback={this.confirm} />
+                </Modal>
+                <AppHeader icon='store' backLink="/" menuName="APPS"/>
+                {this.props.apps ? <HScrollMenu data={this.props.apps.map((app) => {
+                        return {
+                                image: app.iconUrl,
+                                appID: app.policyAppID,
+                                cmdID: app.package_url,
+                                name: app.name
+                            }
+                        })}
+                        theme={this.props.theme}
+                        onSelection={this.onSelection}
+                        isPerformingInteraction={false} /> : null}
+                <AppServices />
+            </div>
+        )
+    }
+}
+
+const mapStateToProps = (state) => {
+    var props = {
+        theme: state.theme,
+        apps: state.appStore.availableApps
+    }
+
+    if (state.appStore.installedApps && state.appStore.availableApps) {
+        props.apps = props.apps.filter((app) => {
+            return !state.appStore.installedApps.find((iApp) => {
+                return app.policyAppID === iApp.policyAppID;
+            });
+        });
+    }
+
+    return props;
+}
+
+export default connect(
+    mapStateToProps
+)(AppStore)
```

Comment:
add line break at end

Reply:
added in https://github.com/smartdevicelink/generic_hmi/pull/240/commits/b79ab73fcfd760a30437c25d53d1c6665ee29526